src/Oscoin/Data/Core.hs

```diff
@@ -0,0 +1,502 @@
+defaultCheckpoint :: Ord (Crypto.Hash c) => Checkpoint' c
+defaultCheckpoint _ = allowAnyMaintainer
+
+-------------------------------------------------------------------------------
+-- Unregister
+-------------------------------------------------------------------------------
+
+type Unregister' c =
+       HasDigitalSignature c
+    => Project c                  -- ^ Project data
+    -> Member c                   -- ^ Transaction signer
+    -> Either Error ()
+
+defaultUnregister :: Ord (Crypto.Hash c) => Unregister' c
+defaultUnregister = allowAnyMaintainer
+
+-------------------------------------------------------------------------------
+-- Authorize
+-------------------------------------------------------------------------------
+
+type Authorize' c =
+       HasDigitalSignature c
+    => Address c                    -- ^ Key to be added
+    -> Project c                    -- ^ Project data
+    -> Member c                     -- ^ Transaction signer
+    -> Either Error ()
+
+defaultAuthorize :: Ord (Crypto.Hash c) => Authorize' c
+defaultAuthorize _ = allowAnyMaintainer
+
+-------------------------------------------------------------------------------
+-- Deauthorize
+-------------------------------------------------------------------------------
+
+type Deauthorize' c =
+       HasDigitalSignature c
+    => Address c                  -- ^ Key to be removed
+    -> Project c                  -- ^ Project data
+    -> Member c                   -- ^ Transaction signer
+    -> Either Error ()
+
+defaultDeauthorize :: Ord (Crypto.Hash c) => Deauthorize' c
+defaultDeauthorize _ = allowAnyMaintainer
+
+-------------------------------------------------------------------------------
+-- UpdateContract
+-------------------------------------------------------------------------------
+
+type UpdateContract' c =
+       HasDigitalSignature c
+    => Handler                  -- ^ Handler to be updated
+    -> HandlerParams            -- ^ New handler parameters
+    -> Project c                -- ^ Project data
+    -> Member c                 -- ^ Transaction signer
+    -> Either Error ()
+
+defaultUpdateContract :: Ord (Crypto.Hash c) => UpdateContract' c
+defaultUpdateContract _ _ = allowAnyMaintainer
+
+-------------------------------------------------------------------------------
+-- Utility functions
+-------------------------------------------------------------------------------
+
+-- | Distribute a balance equally to a set of address, returning the remainder.
+distribute :: Balance -> Set (Address c) -> ([(Address c, Balance)], Balance)
+distribute bal accs =
+    ([(acc, share) | acc <- Set.toList accs], remainder)
+  where
+    share     = bal `div` fromIntegral (Set.size accs)
+    remainder = bal `mod` share
+
+-- | Return 'Right ()' if the member is a project maintainer.
+allowAnyMaintainer :: Ord (Crypto.Hash c) => Project c -> Member c -> Either Error ()
+allowAnyMaintainer Project{..} Member{..} =
```

Comment:
What do you think about calling this `ensureMaintainer` or `requireMaintainer`?

Reply:
:thinking: I like `requireMaintainer`